sklearn/model_selection/_split.py

```diff
@@ -284,7 +284,16 @@ def __init__(self, n_splits, shuffle, random_state):
 
         self.n_splits = n_splits
         self.shuffle = shuffle
+        # For repr
         self.random_state = random_state
+        if random_state is None:
+            # This is done to ensure that the multiple calls to split
+            # are random for each initialization of splitter but consistent
+            # across multiple calls for the same initialization.
+            self._random_state = check_random_state(
+                random_state).randint(np.iinfo(np.int32).max)
```

Comment:
Gaël had instead suggested the use of `get_state`. Apart from the ability to represent many more than 2**32 possible states, it is faster to set the state than to seed a new `RandomState`.

I'm okay with this though.